### PR TITLE
Issue 452 - fixed an issue where parser does not parse the file properly

### PR DESCRIPTION
**Pull request by smitshah91**
fixed an issue where parser does not parse the file properly

1. When there is a quote at the end of the line and there is some extra spaces after closing quote and newline
2. When there is a quote before delimiter and there is some extra spaces after closing quote and next delimiter

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -1285,21 +1285,25 @@
 							continue;
 						}
 
-						// Closing quote followed by delimiter
-						if (input[quoteSearch+1] === delim)
+						var spacesBetweenQuoteAndDelimiter = extraSpacesLengthAfterQuote(nextDelim);
+
+						// Closing quote followed by delimiter or 'unnecessary steps + delimiter'
+						if (input[quoteSearch+1] === delim  || spacesBetweenQuoteAndDelimiter > 0)
```
Comment (pokoli) [inline papaparse.js]: Why not summing the spaces to the QuoteSearch?

File: papaparse.js [hunk 2]
```diff
@@ -1285,21 +1285,25 @@
 							continue;
 						}
 
-						// Closing quote followed by delimiter
-						if (input[quoteSearch+1] === delim)
+						var spacesBetweenQuoteAndDelimiter = extraSpacesLengthAfterQuote(nextDelim);
```
Comment (pokoli) [inline papaparse.js]: Probably we can find a shorter name. Why not extraSpaces?

File: papaparse.js [hunk 3]
```diff
@@ -1285,21 +1285,25 @@
...
-							cursor = quoteSearch + 1 + delimLen;
+							cursor = quoteSearch + 1 + spacesBetweenQuoteAndDelimiter + delimLen;
 							nextDelim = input.indexOf(delim, cursor);
 							nextNewline = input.indexOf(newline, cursor);
 							break;
 						}
 
-						// Closing quote followed by newline
-						if (input.substr(quoteSearch+1, newlineLen) === newline)
+						var spacesBetweenQuoteAndNewLine = extraSpacesLengthAfterQuote(nextNewline);
```
Comment (pokoli) [inline papaparse.js]: idem about name

File: papaparse.js [hunk 4]
```diff
@@ -1285,21 +1285,25 @@
...
+
+						// Closing quote followed by newline or 'unnecessary spaces + newLine'
+						if (input.substr(quoteSearch+1, newlineLen) === newline || spacesBetweenQuoteAndNewLine > 0)
```
Comment (pokoli) [inline papaparse.js]: Idem about sum

File: papaparse.js [hunk 5]
```diff
@@ -1385,6 +1389,21 @@
 				lastCursor = cursor;
 			}
 
+			/**
+             * checks if there are extra spaces after closing quote and newLine without any text
+             * if Yes, returns the number of spaces
+            */
+            function extraSpacesLengthAfterQuote(index) {
+                var spaceLength = 0;
+				if (index !== -1) {
```
Comment (pokoli) [inline papaparse.js]: Which caracter are you using for ident? It not rendered correctly on github, but you should use Tabs to ident code. I will expect this line to be aligned with the one upper.

File: papaparse.js (hunk 2 above)
Comment (smitshah91) [reply papaparse.js]: Agree. was just trying to be specific.

File: papaparse.js (hunk 1 above)
Comment (smitshah91) [reply papaparse.js]: Good point.

File: papaparse.js (hunk 5 above)
Comment (smitshah91) [reply papaparse.js]: Using tabs but IDE might have messed it up. will fix it.

**Pull request closed without merging**